src/Open3D/Visualization/Rendering/Filament/TriangleMeshBuffers.cpp

```diff
@@ -364,108 +364,107 @@ RenderableManager::PrimitiveType TriangleMeshBuffersBuilder::GetPrimitiveType()
         }
 
         // Converting normals to Filament type - quaternions
-        const size_t tangentsBytesCount = nVertices * 4 * sizeof(float);
-        float4VTangents = static_cast<math::quatf*>(malloc(tangentsBytesCount));
+        const size_t tangents_byte_count = n_vertices * 4 * sizeof(float);
+        float4v_tangents =
+                static_cast<math::quatf*>(malloc(tangents_byte_count));
         auto orientation = filament::geometry::SurfaceOrientation::Builder()
-                                   .vertexCount(nVertices)
+                                   .vertexCount(n_vertices)
                                    .normals(reinterpret_cast<math::float3*>(
                                            normals.data()))
                                    .build();
-        orientation.getQuats(float4VTangents, nVertices);
+        orientation.getQuats(float4v_tangents, n_vertices);
     } else {
         utility::LogWarning(
                 "Trying to create mesh without vertex normals. Shading would "
                 "not work correctly. Consider to generate vertex normals "
                 "first.");
     }
 
-    // We default to using vertex color attribute for all geometries, even if
-    // a geometry doesn't have one. That's all due to our default material and
-    // large variety of geometries it should support
-    const bool hasColors = true;  // geometry_.HasVertexColors();
-    const bool hasUVs = geometry_.HasTriangleUvs();
+    // NOTE: Both default lit and unlit material shaders require per-vertex
+    // colors so we unconditionally assume the triangle mesh has color.
+    const bool has_colors = true;
+    const bool has_uvs = geometry_.HasTriangleUvs();
 
     // We take ownership of vbdata.bytes and ibdata.bytes here.
-    std::tuple<vbdata, ibdata> buffersData;
-    if (hasUVs) {
-        buffersData = CreateTexturedBuffers(float4VTangents, geometry_);
-    } else if (hasColors) {
-        buffersData = CreateColoredBuffers(float4VTangents, geometry_);
+    std::tuple<vbdata, ibdata> buffers_data;
+    if (has_uvs) {
+        buffers_data = CreateTexturedBuffers(float4v_tangents, geometry_);
+    } else if (has_colors) {
+        buffers_data = CreateColoredBuffers(float4v_tangents, geometry_);
     } else {
-        buffersData = CreatePlainBuffers(float4VTangents, geometry_);
+        buffers_data = CreatePlainBuffers(float4v_tangents, geometry_);
     }
 
-    free(float4VTangents);
+    free(float4v_tangents);
 
-    const vbdata& vertexData = std::get<0>(buffersData);
-    const ibdata& indexData = std::get<1>(buffersData);
+    const vbdata& vertex_data = std::get<0>(buffers_data);
+    const ibdata& index_data = std::get<1>(buffers_data);
 
     size_t stride = sizeof(BaseVertex);
     VertexBuffer* vbuf = nullptr;
-    if (hasUVs) {
+    if (has_uvs) {
         stride = sizeof(TexturedVertex);
-    } else if (hasColors) {
+    } else {
         stride = sizeof(ColoredVertex);
     }
```

Comment:
can we move setting up stride to be together with calls to CreateTexturedBuffers, CreateColoredBuffers, and CreatePlainBuffers, above, so we can guarantee they always match?

Reply:
changed.